### ISSUE TITLE
actionUtil.parseModel intermittently returns a Model with undefined create method (v0.11.0)

### ISSUE THREAD

**Issue by sofiaguyang** [balderdashy/sails#3162]
Intermittently, my integration test, which I wrote using supertest, is failing with Uncaught TypeError: undefined is not a function at Array.async.auto.createStep (api/blueprint-templates/createTemplate.js:44:13). I am using Sails v0.11.0.

```
 1)  "before each" hook:
     Uncaught TypeError: undefined is not a function
      at Array.async.auto.createStep (api/blueprint-templates/createTemplate.js:44:13)
```

![model create](https://cloud.githubusercontent.com/assets/4891894/9355869/628d35b2-46af-11e5-8011-e4cb23aa0cbc.png)

createTemplate.js is based on create.js but is intended to be called from a controller action. The createTemplate.js accepts a preCreateOp and postCreateOp to be invoked in a pre-create step and a post-create step respectively.

Below is the complete source code for the createTemplate.js:

```javascript
/**
 * Module dependencies
 */
var actionUtil = require('sails/lib/hooks/blueprints/actionUtil.js');


/**
 * Create Record
 *
 * post /:modelIdentity
 *
 * An API call to find and return a single model instance from the data adapter
 * using the specified criteria.  If an id was specified, just the instance with
 * that unique id will be returned.
 *
 * Optional:
 * @param {String} callback - default jsonp callback param (i.e. the name of the js function returned)
 * @param {*} * - other params will be used as `values` in the create
 */
module.exports = function createRecord(req, res, preCreateOp, postCreateOp) {
  var Model = actionUtil.parseModel(req);

  /*while (!(Model.create)) {
    Model = actionUtil.parseModel(req);
    console.log('while(!(Model.create)): ' + Model);
    console.log(Model);
  }*/

  async.auto({
    preCreateStep : function (callback) {
      var data = actionUtil.parseValues(req);

      if (preCreateOp) {
        preCreateOp(data, callback);
      } else {
        callback(null, data);
      }
    },
    createStep    : ['preCreateStep', function (callback, results) {

      var data = results.preCreateStep;

      // Create new instance of model using data from params
      Model.create(data).exec(function created(err, newInstance) {

        // Differentiate between waterline-originated validation errors
        // and serious underlying issues. Respond with badRequest if a
        // validation error is encountered, w/ validation info.
        if (err) {
          return res.negotiate(err);
        }

        callback(null, newInstance);
      });
    }],
    postCreateStep: ['createStep', function (callback, results) {
      var newInstance = results.createStep;

      if (postCreateOp) {
        postCreateOp(newInstance, callback);
      } else {
        callback(null, newInstance);
      }
    }]
  }, function main(callback, results) {
    var newInstance = results.postCreateStep;

    // If we have the pubsub hook, use the model class's publish method
    // to notify all subscribers about the created item
    if (req._sails.hooks.pubsub) {
      if (req.isSocket) {
        Model.subscribe(req, newInstance);
        Model.introduce(newInstance);
      }
      Model.publishCreate(newInstance, !req.options.mirror && req);
    }

    res.created(newInstance.toJSON());
  });
};
```

And below is a sample usage:

```javascript
/**
 * PartyController
 *
 * @description :: Server-side logic for managing Parties
 * @help        :: See http://sailsjs.org/#!/documentation/concepts/Controllers
 */

var uuid = require('node-uuid');
var findTemplate = require('../blueprint-templates/findTemplate');
var findOneTemplate = require('../blueprint-templates/findOneTemplate');
var createTemplate = require('../blueprint-templates/createTemplate');
var _ = require('lodash');

module.exports = {
  create: function (req, res) {
    createTemplate(req, res,
      function validate(data, callback) {
        Party.validate(data, function (err) {
          if (err) {
            return res.json({
              error           : "validation_error",
              errorDescription: "Validation error!"
            });
          } else {
            data.partyId = uuid.v4();
            callback(err, data);
          }
        });
      },
      function addLinks(newInstance, callback) {
        createLinkToSelf(newInstance, req);
        createLinkToKeys(newInstance, req);
        callback(null, newInstance);
      });
  }
};

function createLinkToSelf(party, req) {
  initPartyLinks(party);
  party._links.self = req.protocol + '://' + req.host + ':' + req.port + '/parties/' + party.partyId;
}

function createLinkToKeys(party, req) {
  initPartyLinks(party);
  party._links.keys = req.protocol + '://' + req.host + ':' + req.port + '/parties/' + party.partyId + '/keys';
}

function initPartyLinks(party) {
  if (!party._links) {
    party._links = {};
  }
}

```

Below is my bootstrap.test.js:

```javascript
var Sails = require('sails'),
    sails;

before(function (done) {

  // Increase the Mocha timeout so that Sails has enough time to lift.
  this.timeout(5000);

  Sails.lift({
    environment: 'test',
    port       : 1340,
    log        : {
      level: 'error'
    },
    models: {
      connections: {
        testDB: {
          adapter: 'sails-memory'
        }
      },

      //wipe/drop ALL my data and rebuild models every time
      migrate: 'drop'
    }
  }, function (err, server) {
    sails = server;
    if (err) return done(err);
    // here you can load fixtures, etc.
    done(err, sails);
  });
});

after(function (done) {
  // here you can clear fixtures, etc.
  Sails.lower(done);
});

beforeEach(function (done) {
  // Drops database between each test.  This works because we use the memory database
  sails.once('hook:orm:reloaded', done);
  sails.emit('hook:orm:reload');
});
```

Is there something I am missing or did wrong?

**Comment by sailsbot** [comment]
Thanks for posting, @sofiaguyang.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
